lib/chef/knife/bootstrap.rb

```diff
@@ -246,7 +247,19 @@ def default_bootstrap_template
       #
       # @return [String] The DNS or IP that bootstrap will connect to
       def server_name
-        Array(@name_args).first
+        host_descriptor[:host]
+      end
+
+      def host_descriptor
+        unless @name_args.nil?
+          parts = Array(@name_args).first.split('@')
+          @host_descriptor = {
+            :host => parts.pop,
+            :user => parts.pop
+          }
+        end
+
+        @host_descriptor
```

Comment:
ah, the first pop is the host, regardless of "@", without which the second (user) will be nil. but perhaps that's too unspoken

ie.

```rb
"example.com".split('@').pop
```